sdcm/cluster.py

```diff
@@ -230,7 +229,7 @@ def __init__(self, name, parent_cluster, ssh_login_info=None, base_logdir=None,
         self.logdir = os.path.join(base_logdir, self.name) if base_logdir else None
         self.dc_idx = dc_idx
 
-        self.argus_resource: CloudResource | None = None
+        self.argus_resource = False
```

Comment:
So this value was previously used when terminating a resource to decide if it needs to be terminated or not, since we don't really have state anymore we could just remove it if we're okay with an API exception in the log when the client tries to delete a non-existent resource.

Reply:
you'll have to explain it again, this flag now tell if we create the resource or not ? 
what are we trying to guard with it and why ? 

the only place I see it's being used is in `update_shards_in_argus`, which if we got there it's perfectly safe to assume we created the resource, anyhow we should be failing the test at that point one way or the other.